daemons/based/based_remote.c

```diff
@@ -617,40 +625,45 @@ authenticate_user(const char *user, const char *passwd)
         goto bail;
     }
 
+    // Check user credentials
     rc = pam_authenticate(pam_h, 0);
     if (rc != PAM_SUCCESS) {
         crm_err("Authentication failed for %s: %s (%d)", user, pam_strerror(pam_h, rc), rc);
         goto bail;
     }
 
-    /* Make sure we authenticated the user we wanted to authenticate.
-     * Since we also run as non-root, it might be worth pre-checking
-     * the user has the same EID as us, since that the only user we
-     * can authenticate.
+    /* Get the authenticated user name (PAM modules can map the original name to
+     * something else)
      */
     rc = pam_get_item(pam_h, PAM_USER, &p_user);
     if (rc != PAM_SUCCESS) {
         crm_err("Internal PAM error: %s (%d)", pam_strerror(pam_h, rc), rc);
         goto bail;
-
-    } else if (p_user == NULL) {
+    }
+    if (p_user == NULL) {
         crm_err("Unknown user authenticated.");
         goto bail;
+    }
 
-    } else if (!pcmk__str_eq(p_user, user, pcmk__str_casei)) {
+    // @TODO Why do we require these to match?
+    if (!pcmk__str_eq(p_user, user, pcmk__str_casei)) {
         crm_err("User mismatch: %s vs. %s.", (const char *)p_user, (const char *)user);
         goto bail;
     }
 
+    // Check user account restrictions (expiration, etc.)
     rc = pam_acct_mgmt(pam_h, 0);
     if (rc != PAM_SUCCESS) {
         crm_err("Access denied: %s (%d)", pam_strerror(pam_h, rc), rc);
         goto bail;
     }
-    pass = TRUE;
+    pass = true;
 
-  bail:
+bail:
     pam_end(pam_h, rc);
-#endif
     return pass;
+#else
+    // @TODO Implement for non-PAM environments
+    return true;
```

Comment:
Just curious... what non-PAM environments do we support?

Reply:
*BSD can install OpenPAM but don't have to